### PR TITLE
Bump gateway to 0.6.0

### PR DESCRIPTION
**Pull request by howardjohn**
will remove hold after https://github.com/istio/istio/pull/42545 is merged since I depend on that

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -100,7 +100,7 @@ require (
 	k8s.io/kubectl v0.26.0
 	k8s.io/utils v0.0.0-20221128185143-99ec85e7a448
 	sigs.k8s.io/controller-runtime v0.13.1-0.20221209200815-1e773340c783
-	sigs.k8s.io/gateway-api v0.6.0-rc1
+	sigs.k8s.io/gateway-api v0.6.0-rc2
```
Comment (hzxuzhonghu) [inline go.mod:103]: I do think we should depend on the official v0.6.0 release instead of rc

Comment (howardjohn) [reply go.mod:103]: v0.6.0 is out,. updating

**Review by hzxuzhonghu — APPROVED** [review]

**Review by jacob-delgado — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Review by zhlsunshine — APPROVED** [review]
LGTM

**Pull request merged**